GowerMetric.py

```diff
@@ -154,16 +230,88 @@ def fit(self, X):
                     ratio_cols[:, zero_values_mask], axis=0
                 )
 
-            if self.ratio_scale_normalization == "kde":
-                n = X.shape[0]
-                c = 1.06
-
-                s = np.std(ratio_cols, axis=0)
-                self.h_ = (
-                    c / n ** (1 / 5) * np.min([s, self.ranges_ / 1.34], axis=0)
-                )
-
-        loader = GowerMetricWeights(self)
+            n = X.shape[0]
+
+            # h_t parameter
+            if self.ratio_scale_window == "kde":
+                if self.kde_type == "silverman":
+                    c = 1.06
+
+                    s = np.std(ratio_cols, axis=0)
+                    self.h_ = (
+                        c
+                        / n ** (1 / 5)
+                        * np.min([s, self.ranges_ / 1.34], axis=0)
+                    )
+                elif self.kde_type == "scott":
+                    c = 0.9
+
+                    s = np.std(ratio_cols, axis=0)
+                    self.h_ = c / n ** (1 / 5) * s
+                elif self.kde_type == "sheather-jones":
+                    from KDEpy import FFTKDE
+
+                    try:
+                        self.h_ = np.array(
+                            [
+                                FFTKDE(kernel="gaussian", bw="ISJ")
+                                .fit(ratio_cols[:, i])
+                                .bw
+                                for i in range(self.ratio_scale_num)
+                            ]
+                        )
+                    except ValueError:
+                        print(
+                            "FFTKDE resulted in failure - "
+                            "Crashed with error: ValueError: Root finding did not converge. Need more data. - "
+                            "Repeating fit using kde_type == 'silverman'"
+                        )
+
+                        # Try to fit using silverman instead
+                        self.kde_type = "silverman"
+                        self.fit(X)
+                        return
```

Comment:
This will resort to Silverman's method in case that any variables fails to converge, while it would fall back only for variables that caused the error. Rewrite this part:
1. Use regular for loop with append
2. try/except for each calculation
3. Print message about failure only if `verbose` is above zero. Print it after the loop, and print the number of variables that failed.
4. You do not know that this was an error, since you catch all ValueErrors here. Catch `ValueError as e` instead, and check if `"Root finding did not converge:` is in the error message - only then you can be sure. If error is of other type, reraise it with `raise`